executor/src/witgen/machines/block_machine.rs

```diff
@@ -232,6 +237,7 @@ fn detect_connection_type_and_block_size<'a, T: FieldElement>(
 fn try_to_period<T: FieldElement>(
     expr: &Option<Expression<T>>,
     fixed_data: &FixedData<T>,
+    degree: DegreeType,
```

Comment:
No need to pass the degree here either. We can get it in line 254

Reply:
Do you mean that we should get the common degree from the expression?